examples/imagenet.py

```diff
@@ -0,0 +1,363 @@
+import os
+import warnings
+
+import hydra
+import torch
+import torch.nn as nn
+import torch.backends.cudnn as cudnn
+import torch.distributed as dist
+import torchvision.models as models
+import torchvision.datasets as datasets
+import torchvision.transforms as transforms
+from torch.optim import Adam
+
+best_acc1 = 0
+
+
+@hydra.main(config_name="imagenetconf")
+def main(cfg):
+    if cfg.seed is not None:
+        random.seed(cfg.seed)
+        torch.manual_seed(cfg.seed)
+        cudnn.deterministic = True
+        warnings.warn(
+            "You have chosen to seed training. "
+            "This will turn on the CUDNN deterministic setting, "
+            "which can slow down your training considerably! "
+            "You may see unexpected behavior when restarting "
+            "from checkpoints."
+        )
+    if cfg.gpu is not None:
+        warnings.warn(
+            "You have chosen a specific GPU. This will completely "
+            "disable data parallelism."
+        )
+    if cfg.dist_url == "env://" and cfg.world_size == -1:
+        cfg.world_size = int(os.environ["WORLD_SIZE"])
+
+    global best_acc1
```

Comment:
Why global? is this from an existing example?

Reply:
This is from https://github.com/pytorch/examples/tree/master/imagenet. I can change that part, as I'm not the biggest fan as well. But I decided to basically transcribe what they have since it's a really well established script.

Reply:
ok.

Reply:
Maybe we include two example? `_00` which is basically 'transcribed, with hydra' and `_01` which is 'improved, with hydra'?

This has been my approach in the tutorials so far.

Also, I think this is worth thinking about more closely:
https://github.com/pytorch/hydra-torch/pull/30#issuecomment-719759425